SDWebImage/SDWebImageDownloaderOperation.m

```diff
@@ -349,7 +349,10 @@ - (void)URLSession:(NSURLSession *)session dataTask:(NSURLSessionDataTask *)data
     double currentProgress = (double)self.receivedSize / (double)self.expectedSize;
     double previousProgress = self.previousProgress;
     // Check if we need callback progress
-    if (currentProgress - previousProgress < self.minimumProgressInterval) {
+    if ((currentProgress - previousProgress) < self.minimumProgressInterval || finished) {
```

Comment:
I guess maybe this should be changed to this, right ? The final progress block does not called, which means, if the `minimumProgressInterval = 0.50001`, we should call progressBlock twice (ignore the other progressBlock outsize `URLSession:dataTask:didReceiveData:`).

So, the correct one should be this:

```objective-c
if ((currentProgress - previousProgress) < self.minimumProgressInterval && !finished) {
  // return;
}
```

I can fix this issue, because the bug was introduce by me...

Reply:
Your are correct @dreampiggy, this condition is about skipping the progress block call. So we want to skip when there is too little progress and the download is not finished. If we have enough progress or we are finished, we want to call the progress block.

Reply:
@bpoplauschi Please see #2432 , we would fix this in that `PR`. 😂